File: android/app/src/main/java/com/expensetracker/ui/screens/WelcomeScreen.kt
package com.expensetracker.ui.screens

import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.expensetracker.ui.theme.*
import com.expensetracker.ui.components.*
import kotlinx.coroutines.launch

data class WelcomeFeature(
    val title: String,
    val description: String,
    val icon: ImageVector
)

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun WelcomeScreen(
    onCompleteWelcome: () -> Unit = {}
) {
    val features = listOf(
        WelcomeFeature(
            title = "Track Your Expenses",
            description = "Easily track all your family's income and expenses in one place. Categorize transactions and see where your money goes.",
            icon = Icons.Default.Receipt
        ),
        WelcomeFeature(
            title = "Set Budgets & Goals",
            description = "Create monthly budgets for different categories and set savings goals. Get alerts when you're approaching your limits.",
            icon = Icons.Default.TrendingUp
        ),
        WelcomeFeature(
            title = "Manage Recurring Transactions",
            description = "Set up recurring income and expenses like salary, rent, and subscriptions. Never miss a payment again.",
            icon = Icons.Default.Autorenew
        ),
        WelcomeFeature(
            title = "Family Collaboration",
            description = "Invite family members to contribute to your shared budget. Everyone can add transactions and view reports.",
            icon = Icons.Default.Group
        ),
        WelcomeFeature(
            title = "Detailed Reports",
            description = "Get insights with comprehensive reports and visualizations. Export data to PDF or CSV for your records.",
            icon = Icons.Default.Analytics
        )
    )
    
    val pagerState = rememberPagerState(pageCount = { features.size })
    val coroutineScope = rememberCoroutineScope()
    
    Scaffold(
        containerColor = SurfaceBackground
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(24.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Skip button
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.End
            ) {
                TextButton(
                    onClick = onCompleteWelcome
                ) {
                    Text(
                        "Skip",
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                    )
                }
            }
            
            Spacer(modifier = Modifier.height(32.dp))
            
            // Pager
            HorizontalPager(
                state = pagerState,
                modifier = Modifier.weight(1f)
            ) { page ->
                WelcomeFeaturePage(features[page])
            }
            
            Spacer(modifier = Modifier.height(32.dp))
            
            // Page indicators
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                repeat(features.size) { index ->
                    Box(
                        modifier = Modifier
                            .size(8.dp)
                            .clip(CircleShape)
                            .background(
                                if (index == pagerState.currentPage) PrimaryBlue 
                                else MaterialTheme.colorScheme.onSurface.copy(alpha = 0.3f)
                            )
                    )
                }
            }
            
            Spacer(modifier = Modifier.height(32.dp))
            
            // Navigation buttons
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                if (pagerState.currentPage > 0) {
                    SecondaryButton(
                        text = "Previous",
                        onClick = {
                            coroutineScope.launch {
                                pagerState.animateScrollToPage(pagerState.currentPage - 1)
                            }
                        }
                    )
                } else {
                    Spacer(modifier = Modifier.width(100.dp))
                }
                
                if (pagerState.currentPage < features.size - 1) {
                    PrimaryButton(
                        text = "Next",
                        onClick = {
                            coroutineScope.launch {
                                pagerState.animateScrollToPage(pagerState.currentPage + 1)
                            }
                        }
                    )
                } else {
                    PrimaryButton(
                        text = "Get Started",
                        onClick = onCompleteWelcome
                    )
                }
            }
        }
    }
}

@Composable
private fun WelcomeFeaturePage(feature: WelcomeFeature) {
    Column(
        modifier = Modifier.fillMaxSize(),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        // Feature icon
        Box(
            modifier = Modifier
                .size(120.dp)
                .clip(CircleShape)
                .background(PrimaryBlue.copy(alpha = 0.1f)),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = feature.icon,
                contentDescription = null,
                modifier = Modifier.size(60.dp),
                tint = PrimaryBlue
            )
        }
        
        Spacer(modifier = Modifier.height(48.dp))
        
        // Feature title
        Text(
            text = feature.title,
            style = MaterialTheme.typography.headlineLarge,
            color = MaterialTheme.colorScheme.onSurface,
            textAlign = TextAlign.Center
        )
        
        Spacer(modifier = Modifier.height(24.dp))
        
        // Feature description
        Text(
            text = feature.description,
            style = MaterialTheme.typography.bodyLarge,
            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f),
            textAlign = TextAlign.Center,
            lineHeight = MaterialTheme.typography.bodyLarge.lineHeight * 1.4
        )
    }
}

@Preview(showBackground = true)
@Composable
fun WelcomeScreenPreview() {
    ExpenseTrackerTheme {
        WelcomeScreen()
    }
}

File: ios/ExpenseTracker/Screens/WelcomeScreen.swift
import SwiftUI

struct WelcomeFeature {
    let title: String
    let description: String
    let iconName: String
}

struct WelcomeScreen: View {
    @State private var currentPage = 0
    let onCompleteWelcome: () -> Void
    
    private let features = [
        WelcomeFeature(
            title: "Track Your Expenses",
            description: "Easily track all your family's income and expenses in one place. Categorize transactions and see where your money goes.",
            iconName: "receipt"
        ),
        WelcomeFeature(
            title: "Set Budgets & Goals",
            description: "Create monthly budgets for different categories and set savings goals. Get alerts when you're approaching your limits.",
            iconName: "chart.line.uptrend.xyaxis"
        ),
        WelcomeFeature(
            title: "Manage Recurring Transactions",
            description: "Set up recurring income and expenses like salary, rent, and subscriptions. Never miss a payment again.",
            iconName: "arrow.clockwise"
        ),
        WelcomeFeature(
            title: "Family Collaboration",
            description: "Invite family members to contribute to your shared budget. Everyone can add transactions and view reports.",
            iconName: "person.3"
        ),
        WelcomeFeature(
            title: "Detailed Reports",
            description: "Get insights with comprehensive reports and visualizations. Export data to PDF or CSV for your records.",
            iconName: "chart.bar"
        )
    ]
    
    var body: some View {
        VStack(spacing: 0) {
            // Skip button
            HStack {
                Spacer()
                
                Button("Skip") {
                    onCompleteWelcome()
                }
                .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                .padding()
            }
            
            // Content
            TabView(selection: $currentPage) {
                ForEach(features.indices, id: \.self) { index in
                    WelcomeFeaturePage(feature: features[index])
                        .tag(index)
                }
            }
            .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
            .animation(.easeInOut, value: currentPage)
            
            // Page indicators
            HStack(spacing: Theme.Spacing.xs) {
                ForEach(features.indices, id: \.self) { index in
                    Circle()
                        .fill(index == currentPage ? Theme.Colors.primary : Theme.Colors.onSurface.opacity(0.3))
                        .frame(width: 8, height: 8)
                        .animation(.easeInOut, value: currentPage)
                }
            }
            .padding(.vertical, Theme.Spacing.lg)
            
            // Navigation buttons
            HStack {
                if currentPage > 0 {
                    SecondaryButton(title: "Previous") {
                        withAnimation {
                            currentPage -= 1
                        }
                    }
                } else {
                    Spacer()
                        .frame(width: 100)
                }
                
                Spacer()
                
                if currentPage < features.count - 1 {
                    PrimaryButton(title: "Next") {
                        withAnimation {
                            currentPage += 1
                        }
                    }
                } else {
                    PrimaryButton(title: "Get Started") {
                        onCompleteWelcome()
                    }
                }
            }
            .padding(.horizontal, Theme.Spacing.lg)
            .padding(.bottom, Theme.Spacing.lg)
        }
        .background(Theme.Colors.surface)
    }
}

struct WelcomeFeaturePage: View {
    let feature: WelcomeFeature
    
    var body: some View {
        VStack(spacing: Theme.Spacing.xl) {
            Spacer()
            
            // Feature icon
            ZStack {
                Circle()
                    .fill(Theme.Colors.primary.opacity(0.1))
                    .frame(width: 120, height: 120)
                
                Image(systemName: feature.iconName)
                    .font(.system(size: 60))
                    .foregroundColor(Theme.Colors.primary)
            }
            
            VStack(spacing: Theme.Spacing.lg) {
                // Feature title
                Text(feature.title)
                    .font(Theme.Typography.titleXL)
                    .foregroundColor(Theme.Colors.onSurface)
                    .multilineTextAlignment(.center)
                
                // Feature description
                Text(feature.description)
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                    .multilineTextAlignment(.center)
                    .lineSpacing(4)
                    .padding(.horizontal, Theme.Spacing.lg)
            }
            
            Spacer()
        }
        .padding(Theme.Spacing.lg)
    }
}

#Preview {
    WelcomeScreen {
        // Handle completion
    }
}